src/Datastore/DatastoreClient.php

```diff
@@ -449,30 +451,89 @@ public function allocateIds(array $keys, array $options = [])
     }
 
     /**
-     * Create a Transaction
+     * Create a Transaction.
      *
      * Example:
      * ```
      * $transaction = $datastore->transaction();
      * ```
      *
      * @see https://cloud.google.com/datastore/docs/concepts/transactions Datastore Transactions
+     * @see https://cloud.google.com/datastore/docs/reference/rest/v1/projects/beginTransaction beginTransaction
      *
-     * @param array $options [optional] Configuration options.
+     * @codingStandardsIgnoreStart
+     * @param array $options {
+     *     Configuration options.
+     *
+     *      @type string $previous If the transaction is being restarted,
+     *           the previous transaction ID.
+     *     @type array $transactionOptions Any other transaction configuration,
```

Comment:
Is this nested to "future proof" any additional options? I wonder if we could just have any future options live at the same level as `previous`?

Reply:
> Is this nested to "future proof" any additional options?

That's the idea, yes!

> I wonder if we could just have any future options live at the same level as previous?

How would we determine which members of $options are part of $transactionOptions, and which are intended for passthrough to the connection layer? It seems that moving them into a dedicated sub-section provides built-in disambiguation of that point.

Reply:
I understand it is a nice convenience, it just seems a little odd to have the `previous` option live outside of transaction options. What do you think about calling `transactionOptions` `additionalTransactionOptions`?

Reply:
@dwsupplee I updated the code to accept `transactionOptions` in conformance with the API. `previous` is no longer a separate option.